internal/cephfs/nodeserver.go

```diff
@@ -694,6 +711,18 @@ func (ns *NodeServer) NodeGetVolumeStats(
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
+	// health check first, return without stats if unhealthy
+	healthy, msg := ns.healthChecker.IsHealthy(req.GetVolumeId())
+	if !healthy {
+		return &csi.NodeGetVolumeStatsResponse{
```

Comment:
Nothing. The `VolumeCondition` is not used anywhere directly. [Kubernetes can report it](https://kubernetes.io/docs/concepts/storage/volume-health-monitoring/), but does not do so by default (yet).

Reply:
what i mean was if the cephcsi is upgraded we might return the volume as unhealthy for the already mounted pvc because health checker will only gets started during NodeStageVolume? same question for cephcsi restart as well. `IsHealthy` returns error if there is no health check found

Reply:
this is actually addressed in the latest revision of this PR too